python/lsst/dax/imgserv/locateImage.py

```diff
@@ -139,32 +138,34 @@ def getImageFullWithMetadata(self, ra, dec, filterName):
         metadata = self._getMetadata(butler, res)
         return img, metadata
 
-
-    def getImage(self, ra, dec, filterName, width, height, cutoutType="arcsecond"):
+    def getImage(self, ra, dec, filterName, width, height, cutoutType="arcsecond", dataId=None):
         '''Return an image centered on ra and dec (in degrees) with dimensions
         height and width (in arcseconds).
         Returns None if no image is found.
         This function assumes the entire image is valid. (no overscan, etc.)
         Sequence of events:
-         - Use filterName, ra, dec, width, and height to find an image from the database.
-         - Use the results of the query to get an image and metadata from the butler.
-         - Map ra, dec, width, and height to a box.
-         - If a pixel cutout, trim the dimesions to fit in the source image and return.
-         -     and return the cutout.
-         - Otherwise, the height and width are in arcseconds.
-         - Determine approximate pixels per arcsecond in the image by
+        - If not None, dataId is the image id for the butler, ignore next line.
+        - Use filterName, ra, dec, width, and height to find an image from the database.
+        - Use the results of the query to get an image and metadata from the butler.
+        - Map ra, dec, width, and height to a box.
+        - If a pixel cutout, trim the dimesions to fit in the source image and return.
+        -     and return the cutout.
+        - Otherwise, the height and width are in arcseconds.
+        - Determine approximate pixels per arcsecond in the image by
              calculating the length of line from the upper right corner of
              the image to the lower left corner in pixels and arcseconds.
              (This will fail at or very near the pole.)
-         - Use that to define a box for the cutout.
-         - Trim the box so it is entirely within the source image.
-         - Make and return the cutout.
+        - Use that to define a box for the cutout.
+        - Trim the box so it is entirely within the source image.
+        - Make and return the cutout.
         '''
         self._log.debug("getImage %f %f %f %f", ra, dec, width, height)
-        # Find the nearest image to ra and dec.
-        qresult = self._findNearestImageContaining(ra, dec, filterName)
+        qresult = dataId
+        if qresult is None:
+            # Find the nearest image to ra and dec.
+            qresult = self._findNearestImageContaining(ra, dec, filterName)
```

Comment:
I think it might be better to have a separate function that gets the image from the dataId and one that gets the data from the RA and Dec, which would then call the function that gets the image from the dataId.

My concern is that there'd be a silent failure with an empty dataId as the end user would still get an image, but not the one they were expecting, and it wouldn't be immediately obvious in many cases.